### PR TITLE
[Python] bools - replace about.md file with concept files

### PR DESCRIPTION
**Pull request by ErikSchierboom**
In [this issue](https://github.com/exercism/v3/issues/2293) we're describing an evolution of the specification of this repo.
These changes include, amongst others, replacing the contents of the `after.md` document with individual concept documents. There are two documents per concept:

1. An `about.md` file containing the description of the concept
1. A `links.json` file containing a set of useful links related to the concept. 

This looks as follows in the repo:

```
<track>
├── concepts
│   ├── <concept>
│   │   ├── about.md
│   │   └── links.json
│   └── ...
```

This PR applies this change to this exercise by: 
                
1. Copying the existing `after.md` file's contents to the `about.md` file of each concept listed in the exercise's `concepts` section in the `config.json` file.
1. Extract the links from the `after.md` file and put them into the `links.json` file

Before merging this PR, please check that the contents of the concept documents makes sense.
This is especially true when the exercise unlocks multiple concepts, as then the concept documents should be updated to only contain the information relevant to that concept.

Note: to make reviewing easier, the PR has been split into three separate commits. Please squash this PR upon merging.

### CODE REVIEW

File: languages/python/concepts/bools/links.json [hunk 1]
```diff
@@ -1 +1,22 @@
-[]
+[
+  {
+    "url": "https://docs.python.org/3/library/stdtypes.html#truth",
+    "description": "boolean"
+  },
+  {
+    "url": "https://docs.python.org/3/library/stdtypes.html#truth",
+    "description": "boolean"
+  },
```
Comment (cmccandless) [inline languages/python/concepts/bools/links.json:9]: There are duplicate entries in this file; is this intentional? I'm guessing it's a result of the script used counting reference-style links as separate links... but it was still able to grab the referenced URL though?

@ErikSchierboom, can you clarify the intent here?

Comment (ErikSchierboom) [reply languages/python/concepts/bools/links.json:9]: The script was basically just a dumb scraper. Duplicates should definitely be removed.

File: languages/python/concepts/bools/links.json [hunk 2]
```diff
@@ -1 +1,22 @@
...
+  {
+    "url": "https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not",
+    "description": "boolean-operators"
+  },
+  {
+    "url": "https://docs.python.org/3/library/functions.html#bool",
+    "description": "bool-function"
+  }
+]
```
Comment (BethanyG) [inline languages/python/concepts/bools/links.json:22]:
```suggestion
[
  ```suggestion
[
  {
    "url": "https://docs.python.org/3/library/stdtypes.html#bltin-boolean-values",
    "description": "boolean values"
  },
    {
    "url": "https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not",
    "description": "boolean-operators"
  },
  {
    "url": "https://docs.python.org/3/library/stdtypes.html#truth",
    "description": "Truth Value Testing"
  },
  {
    "url": "https://docs.python.org/3/library/functions.html#bool",
    "description": "bool() function"
  },
  {
    "url": "https://problemsolvingwithpython.com/04-Data-Types-and-Variables/04.02-Boolean-Data-Type/",
    "description": "Problem Solving with Python: Boolean Data Type"
  },
    {
    "url": "https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_true.html",
    "description": "Python Anti-Patterns:  Comparing things to True in the Wrong Way"
  },
  {
    "url": "https://www.python.org/dev/peps/pep-0285/",
    "description": "PEP285: Adding a bool type"
  }
  ]
```

**Review by BethanyG — APPROVED** [review]
This LGTM.  Any further thoughts @cmccandless??

File: languages/python/concepts/bools/links.json [hunk 3]
```diff
@@ -1 +1,32 @@
-[]
+[
+  ```suggestion
```
Comment (BethanyG) [inline languages/python/concepts/bools/links.json:2]:
```suggestion
```

**Review by cmccandless — APPROVED** [review]

File: languages/python/concepts/bools/links.json [hunk 4]
```diff
@@ -1 +1,31 @@
...
+  },
+    {
+    "url": "https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_true.html",
+    "description": "Python Anti-Patterns:  Comparing things to True in the Wrong Way"
+  },
+  {
+    "url": "https://www.python.org/dev/peps/pep-0285/",
+    "description": "PEP285: Adding a bool type"
+  }
+  ]
```
Comment (BethanyG) [inline languages/python/concepts/bools/links.json:31]:
```suggestion
 ]
```

File: languages/python/concepts/bools/links.json [hunk 5]
```diff
@@ -1 +1,31 @@
...
+  },
+    {
+    "url": "https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_true.html",
+    "description": "Python Anti-Patterns:  Comparing things to True in the Wrong Way"
+  },
+  {
+    "url": "https://www.python.org/dev/peps/pep-0285/",
+    "description": "PEP285: Adding a bool type"
+  }
+ ]
```
Comment (BethanyG) [inline languages/python/concepts/bools/links.json:31]:
```suggestion
 ]
 
```

File: languages/python/concepts/bools/links.json [hunk 6]
```diff
@@ -1 +1,32 @@
-[]
+[
+[
```
Comment (BethanyG) [inline languages/python/concepts/bools/links.json:2]:
```suggestion
```

File: languages/python/concepts/bools/links.json [hunk 7]
```diff
@@ -1 +1,31 @@
...
+    "url": "https://docs.python.org/3/library/stdtypes.html#truth",
+    "description": "Truth Value Testing"
+  },
+  {
+    "url": "https://docs.python.org/3/library/functions.html#bool",
+    "description": "bool() function"
+  },
+  {
+    "url": "https://problemsolvingwithpython.com/04-Data-Types-and-Variables/04.02-Boolean-Data-Type/",
+    "description": "Problem Solving with Python: Boolean Data Type"
```
Comment (BethanyG) [inline languages/python/concepts/bools/links.json:20]:
```suggestion
    "description": "Problem Solving with Python - Boolean Data Type"
```

File: languages/python/concepts/bools/links.json [hunk 8]
```diff
@@ -1 +1,31 @@
...
+  },
+    {
+    "url": "https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_true.html",
+    "description": "Python Anti-Patterns:  Comparing things to True in the Wrong Way"
+  },
+  {
+    "url": "https://www.python.org/dev/peps/pep-0285/",
+    "description": "PEP285: Adding a bool type"
+  }
+ ]
```
Comment (BethanyG) [inline languages/python/concepts/bools/links.json:30]:
```suggestion
 {
    "url": "https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_true.html",
    "description": "Python Anti-Patterns  Comparing things to True in the Wrong Way"
  },
  {
    "url": "https://www.python.org/dev/peps/pep-0285/",
    "description": "PEP285 - Adding a bool type"
  }
]
```

**Pull request merged**